### PR TITLE
fix: refactor tech debt in aws and identity pool credentials

### PR DESCRIPTION
**Pull request by aeitzman**
implements fixes suggested in #1484

### CODE REVIEW

File: google/auth/aws.py [hunk 1]
```diff
@@ -690,7 +695,7 @@ def __init__(
                 "regional_cred_verification_url"
             )
 
-            # Get the environment ID. Currently, only one version supported (v1).
+            # Get the environment ID, i.e. "aws1". Currently, only one version supported (1).
```
Comment (BigTailWolf) [inline google/auth/aws.py:698]: Is this change from `v1` to `1` just want to clarify we are using the `env_version` literally `1`?

**Review by BigTailWolf — COMMENTED** [review]
Just a nit question

File: google/auth/aws.py [hunk 2]
```diff
@@ -69,6 +69,8 @@
 _DEFAULT_AWS_REGIONAL_CREDENTIAL_VERIFICATION_URL = (
     "https://sts.{region}.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15"
 )
+# IMDSV2 session token lifetime. This is set to a low value because the session token is used immediately.
+_IMDSV2_SESSION_TOKEN_TTL = "300"
```
Comment (clundin25) [inline google/auth/aws.py:73]: Can you include the unit as well?
```suggestion
_IMDSV2_SESSION_TOKEN_TTL _SECONDS= "300"
```

**Review by clundin25 — APPROVED** [review]

File: google/auth/aws.py (hunk 1 above)
Comment (aeitzman) [reply google/auth/aws.py:698]: Correct, the value we expect for environment ID is "aws1"

**Pull request merged**